src/ca/noxid/lab/mapdata/MapInfo.java

```diff
@@ -921,7 +921,7 @@ public byte getTileB(int x, int y, int layer) {
 	public int getTile(int x, int y, int layer) {
 		int rVal = 0;
 		try {
-			if (EditorApp.EDITOR_MODE == 0) {
+			if (EditorApp.EDITOR_MODE == 0 && layer == -1) {
```

Comment:
not sure why this change was necessary? in what situations will layer be -1?
should the && have been an || maybe?

Reply:
Adding that seems to fix the layer visibility settings (and the equivalent addition in putTile seems to fix undoing/redoing).